content/en/docs/concepts/scheduling-eviction/pod-eviction.md

```diff
@@ -0,0 +1,375 @@
+holding for the defined grace periods. This causes the reported node condition
+to constantly switch between `true` and `false`, leading to bad eviction decisions.
+
+To protect against oscillation, you can use the `eviction-pressure-transition-period`
+flag, which controls how long the kubelet must wait before transitioning a node
+condition to a different state.
+
+## Reclaiming node level resources {#reclaim-node-resources}
+
+The kubelet tries to reclaim node-level resources before it evicts end-user pods.
+
+When a `DiskPressure` node condition is reported, the kubelet reclaims node-level
+resources based on the filesystems on the node. 
+
+### With `imagefs`
+
+If the node has a dedicated `imagefs` filesystem for container runtimes to use,
+the kubelet does the following:
+
+  * If the `nodefs` filesystem meets the eviction threshlds, the kubelet deletes
+    dead pods and containers. 
+  * If the `imagefs` filesystem meets the eviction thresholds, the kubelet
+    deletes all unused images. 
+
+### Without `imagefs`
+
+If the node only has a `nodefs` filesystem that meets eviction thresholds,
+the kubelet frees up disk space in the following order:
+
+1. Delete dead pods and containers
+1. Delete unused images
+
+## Evicting end-user pods
+
+If the kubelet's attempts to reclaim node-level resources don't bring the eviction
+signal below the threshold, the kubelet begins to evict end-user pods. 
+
+The kubelet uses the following parameters to determine pod eviction order:
+
+1. Pod resource usage relative to defined requests
+1. [Pod priority](/docs/concepts/configuration/pod-priority-preemption/)
+1. Resource consumption relative to scheduling requests
+1. Pod [QoS class](/docs/tasks/configure-pod-container/quality-service-pod/)
+
+As a result, kubelet ranks and evicts pods in the following order:
+
+1. `BestEffort` or `Burstable` pods where the usage exceeds requests. These pods
+   are evicted based on their Priority and then by how much their usage level
+   exceeds the request.
+1. `Guaranteed` pods and `Burstable` pods where the usage is less than requests
+   are evicted last, based on their Priority.
+
+`Guaranteed` pods are guaranteed only when requests and limits are specified for
+all the containers and they are equal. These pods will never be evicted because
+of another pod's resource consumption. If a system daemon (such as `kubelet`,
+`docker`, and `journald`) is consuming more resources than were reserved via 
+`system-reserved` or `kube-reserved` allocations, and the node only has
+`Guaranteed` or `Burstable` pods using less resources than requests left on it,
+then the kubelet must choose to evict one of these pods to preserve node stability
+and to limit the impact of resource starvation on other Pods. In this case, it
+will choose to evict pods of lowest Priority first.
+
+When the kubelet evicts pods to reclaim disk resources, it sorts pods by QoS and
```

Comment:
This isn't correct.  It is by the same algorithm as CPU and memory.  Note that there are no requests for inodes or PIDs, so priority ends up being the first criterion.

Reply:
So for disk resources, it uses the same logic as for CPU and memory with the exception of using the QoS class? And then for `inodes` and `PIDs`, only the Priority is used to rank the pods?

Reply:
For disk, inodes, and PIDs, it uses the same logic as for memory.  (There isn't CPU eviction).  It does _not_ ever take QoS into account.  There are no requests for inodes or PIDs, so the algorithm:

1. Usage > request ?
2. Priority
3. Usage - request

Becomes 

1. Usage > 0 ? (always true)
2. Priority
3. Usage - 0 = Usage

For inodes and PIDs.